### PR TITLE
Get bitcoin block hash for the given block number with number of confirmations

### LINKED ISSUE #404: Get bitcoin block hash for the given block number with number of confirmations
- [ ] Define what are parameter and result
- [ ] Bitcoin block info library
- [ ] Logic script to store in chain

### PR DESCRIPTION
**Pull request by prin-r**
Fix #404

### CODE REVIEW

File: owasm/example/bitcoin_info/src/logic.rs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+decl_data! {
+    pub struct Data {
+        pub block_hash: String = |params: &Parameter| block_hash::Info::new(params.block_height),
+        pub block_count: u64 = |_: &Parameter| block_count::Info::new(),
+    }
+}
+
+decl_result! {
+    pub struct Result {
+        pub block_hash: String,
```
Comment (taobun) [inline owasm/example/bitcoin_info/src/logic.rs]: We have string as output how our solidity library need to support this?

File: owasm/src/ext/bitcoin/block_count.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+impl Oracle for Info {
+    type T = u64;
+
+    fn as_cmd(&self) -> ShellCmd {
+        Curl::new(&["https://blockchain.info/q/getblockcount"]).as_cmd()
+    }
+
+    fn from_cmd_output(&self, output: String) -> Option<u64> {
+        json::parse(&output).ok()?.as_u64()
```
Comment (taobun) [inline owasm/src/ext/bitcoin/block_count.rs]: Output is `616081` we need to use json parser?

**Review by taobun — CHANGES_REQUESTED** [review]

File: owasm/example/bitcoin_info/src/logic.rs (hunk 1 above)
Comment (prin-r) [reply owasm/example/bitcoin_info/src/logic.rs]: I will replace this thing by 4 of u64

Comment (prin-r) [reply owasm/example/bitcoin_info/src/logic.rs]: done

File: owasm/src/ext/bitcoin/block_count.rs (hunk 2 above)
Comment (prin-r) [reply owasm/src/ext/bitcoin/block_count.rs]: done

File: owasm/example/bitcoin_info/src/logic.rs [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+            execute(params, vec![data1, data2, data3]),
+            Result {
+                block_hash: [0, 3502949345222752, 16236236487057196107, 2784772046813676667],
+                confirmation: 616058 - 616047
+            }
+        );
+    }
+
+    #[test]
+    fn test_execute_when_the_minority_report_differently() {
```
Comment (taobun) [inline owasm/example/bitcoin_info/src/logic.rs:72]: Where difference here?

**Review by taobun — CHANGES_REQUESTED** [review]

File: owasm/example/bitcoin_info/src/logic.rs (hunk 3 above)
Comment (prin-r) [reply owasm/example/bitcoin_info/src/logic.rs:72]: done

**Review by taobun — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**